src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

There is a potential issue with handling the exception response. The logic for extracting the error message seems convoluted and could be simplified for better readability and maintainability. Consider refactoring this part of the code.

```typescript
// Simplified error message extraction
const error = exception.getResponse()['message'] || 'Internal server error';

// Logging the error message
this.logger.error(`Message: ${error}`, `${context.getRequest().method} ${context.getRequest().url}`);

// Sending a more concise error response
response.status(status).json(APIResponse.create(0, status, error));
```


<!-- This is an auto-generated comment by PullProofAI -->